pkg/admission/webhook.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"k8s.io/api/admission/v1beta1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WebhookType defines the type of the webhook
+type WebhookType int
+
+const (
+	// WebhookTypeMutating represents mutating type webhook
+	WebhookTypeMutating WebhookType = iota
+	// WebhookTypeValidating represents validating type webhook
+	WebhookTypeValidating
+)
+
+// Handler can handle an AdmissionRequest.
+type Handler interface {
+	Handle(ctx context.Context, request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse
```

Comment:
Wondering if we should introduce our own `Request` struct which contains `context` and `AdmissionRequest`. That will help us introducing new fields without breaking any changes.

Similar goes for the the return `AdmissionResponse`.